src/main/java/com/google/cloud/spanner/jdbc/AbstractJdbcPreparedStatement.java

```diff
@@ -81,7 +83,17 @@ public void addBatch(String sql) throws SQLException {
   @Override
   public void setNull(int parameterIndex, int sqlType) throws SQLException {
     checkClosed();
-    parameters.setParameter(parameterIndex, null, sqlType, null);
+    if (sqlType == Types.OTHER || sqlType == Types.NULL) {
+      parameters.setParameter(
+          parameterIndex,
+          Value.untyped(
+              com.google.protobuf.Value.newBuilder().setNullValue(NullValue.NULL_VALUE).build()),
+          sqlType,
+          /* scaleOrLength = */ null);
+    } else {
+      parameters.setParameter(
+          parameterIndex, /* value = */ null, sqlType, /* scaleOrLength = */ null);
+    }
```

Comment:
Instead of adding this untyped logic here, can we do the following?
1.  Add untyped value binding logic under default case of `setNullValue()` method in `JdbcParameterStore.java`, [Code link](https://github.com/googleapis/java-spanner-jdbc/blob/8f75d26c18844617e275feb952db415a067a91d2/src/main/java/com/google/cloud/spanner/jdbc/JdbcParameterStore.java#L928)
2. 
```suggestion
    parameters.setParameter(parameterIndex, null, sqlType, null);
```
3. Also, in [setNullValue()](https://github.com/googleapis/java-spanner-jdbc/blob/8f75d26c18844617e275feb952db415a067a91d2/src/main/java/com/google/cloud/spanner/jdbc/JdbcParameterStore.java#L865) method, currently if `sqlType` is null, we bind it to `String` Value, instead we should change it to use below. Otherwise when we do `setObject(1, null)` and if the column is an array type, then we currently bind it to String which are not compatible and backend throws an error. Instead, if we use untyped value then it works for all cases.
```
 binder.to(Value.untyped(com.google.protobuf.Value.newBuilder().setNullValue(NullValue.NULL_VALUE).build());
```
4. Ideally doing this change in [setNullValue()](https://github.com/googleapis/java-spanner-jdbc/blob/8f75d26c18844617e275feb952db415a067a91d2/src/main/java/com/google/cloud/spanner/jdbc/JdbcParameterStore.java#L865) method, handles all the following cases by binding untyped null value and also works for other methods like `setObject()`.
-  value is null and type is null
-  value is null and type is Types.OTHER, Types.NULL or not known.

Knut, please correct me if something is unclear or not correct here.

Reply:
Thanks, Knut.